### DISCUSSION TITLE
documentation build testability?

### DISCUSSION THREAD

**Discussion by matanox** [mkdocstrings/mkdocstrings#680] [Q&A] [answered]
Is there any (python) api for testing that the docs build without errors? 

I would like to test this as part of my overall project's test suite, hopefully without resorting to shelling out to execute `mkdocs build` as a shell command in a temporary directory, though that's what I'll eventually do if there's no api for the build operation. But that will be hairy and less reliable.

**Comment by rraulf** [comment]
Does it help you?

https://github.com/mkdocs/mkdocs/issues/2186

**Reply by rraulf** [reply]
@matanox Also here there is some information about the build: 
https://www.mkdocs.org/getting-started/

**Reply by rraulf** [reply]
@matanox Actually https://www.mkdocs.org/getting-started/ not relevant for you since it is as a shell command.

**Comment by rraulf** [comment]
@matanox Also here there is some information about the build:
https://www.mkdocs.org/getting-started/ 
in the "Building the site" section:
https://www.mkdocs.org/getting-started/#building-the-site

I hope it will help you.

**Comment by pawamoy** [comment, marked as answer]
You can use this to build your docs in strict mode from Python:

```python
from mkdocs.commands.build import build
from mkdocs.config import load_config

mkdocs_config = load_config()
mkdocs_config["strict"] = True
mkdocs_config["site_dir"] = "/tmp/..."

mkdocs_config["plugins"].run_event("startup", command="build", dirty=False)
try:
    build(mkdocs_config)
finally:
    mkdocs_config["plugins"].run_event("shutdown")
```

**Reply by rraulf** [reply]
Is it running the build twice (in the run_event line and in the try block)?

**Reply by pawamoy** [reply]
No, only once. We explicitly call the `on_startup` and `on_shutdown` hooks here because the `build` function doesn't do it itself. You could try to omit them, but this might break some plugins. Feel free to try with and without :smile:

**Reply by rraulf** [reply]
@pawamoy Thank you!

**Comment by rraulf** [comment]
@matanox I hope my replies helped you anyway.

**Comment by matanox** [comment]
Yes of course, the code for accomplishing it is really much of a solution I hope.

**Reply by rraulf** [reply]
@matanox yes, thank you!

**Reply by rraulf** [reply]
@matanox I'm glad it helped a little.

**Comment by rraulf** [comment]
@matanox I'm glad it helped a little.

**Comment by pawamoy** [comment]
For some reason GitHub keeps notifying me of this issue while nothing changed in it. So I'll close and lock it, if you don't mind :slightly_smiling_face:
